src/Driver/OCI8/OCI8Statement.php

```diff
@@ -219,120 +207,131 @@ public function execute(?array $params = null) : void
-        );
+        return FetchUtils::fetchOneFromNumeric($this);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function fetchAll(?int $fetchMode = null) : array
+    public function fetchAllNumeric() : array
     {
-        $fetchMode = $fetchMode ?? $this->_defaultFetchMode;
-
-        $result = [];
-
-        if (! isset(self::$fetchModeMap[$fetchMode])) {
-            throw new InvalidArgumentException(sprintf('Invalid fetch mode %d.', $fetchMode));
-        }
-
-        if (self::$fetchModeMap[$fetchMode] === OCI_BOTH) {
-            while ($row = $this->fetch($fetchMode)) {
-                $result[] = $row;
-            }
-        } else {
-            $fetchStructure = OCI_FETCHSTATEMENT_BY_ROW;
+        return $this->fetchAll(OCI_NUM, OCI_FETCHSTATEMENT_BY_ROW);
+    }
 
-            if ($fetchMode === FetchMode::COLUMN) {
-                $fetchStructure = OCI_FETCHSTATEMENT_BY_COLUMN;
-            }
+    /**
+     * {@inheritdoc}
+     */
+    public function fetchAllAssociative() : array
+    {
+        return $this->fetchAll(OCI_ASSOC, OCI_FETCHSTATEMENT_BY_ROW);
+    }
 
-            // do not try fetching from the statement if it's not expected to contain result
-            // in order to prevent exceptional situation
-            if (! $this->result) {
-                return [];
-            }
+    /**
+     * {@inheritdoc}
+     */
+    public function fetchColumn() : array
+    {
+        return $this->fetchAll(OCI_NUM, OCI_FETCHSTATEMENT_BY_COLUMN)[0];
+    }
 
-            oci_fetch_all(
-                $this->_sth,
-                $result,
-                0,
-                -1,
-                self::$fetchModeMap[$fetchMode] | OCI_RETURN_NULLS | $fetchStructure | OCI_RETURN_LOBS
-            );
+    public function rowCount() : int
+    {
+        $count = oci_num_rows($this->_sth);
 
-            if ($fetchMode === FetchMode::COLUMN) {
-                $result = $result[0];
-            }
+        if ($count !== false) {
+            return $count;
         }
 
-        return $result;
+        return 0;
     }
 
     /**
-     * {@inheritdoc}
+     * @return mixed|false
      */
-    public function fetchColumn()
+    private function fetch(int $mode)
     {
         // do not try fetching from the statement if it's not expected to contain result
         // in order to prevent exceptional situation
         if (! $this->result) {
             return false;
         }
 
-        $row = oci_fetch_array($this->_sth, OCI_NUM | OCI_RETURN_NULLS | OCI_RETURN_LOBS);
+        return oci_fetch_array(
+            $this->_sth,
+            $mode | OCI_RETURN_NULLS | OCI_RETURN_LOBS
+        );
+    }
 
-        if ($row === false) {
-            return false;
+    /**
+     * @return array<mixed>
+     */
+    private function fetchAll(int $mode, int $fetchStructure) : array
+    {
+        // do not try fetching from the statement if it's not expected to contain result
```

Comment:
```suggestion
        // do not try fetching from the statement if it's not expected to contain results
```

Reply:
It will be “the result” since it refers to the `$result` property or “statement result” in general (like `mysqli_result`).